pandas/tests/frame/test_analytics.py

```diff
@@ -256,6 +256,27 @@ def test_corrwith_mixed_dtypes(self):
         expected = pd.Series(data=corrs, index=['a', 'b'])
         tm.assert_series_equal(result, expected)
 
+    def test_corrwith_how_all(self):
+        df = DataFrame({'a': np.random.randn(10000),
+                        'b': np.random.randn(10000)})
+        c1 = df.corrwith(df, how='all').loc['a', 'b']
+        c2 = np.corrcoef(df['a'], df['b'])[0][1]
+
+        tm.assert_almost_equal(c1, c2)
+        assert c1 < 1
```

Comment:
need a lot more tests here, e.g. to hit all of the error conditions you are now asserting, as well as non-aligned and non-unique frames.

Reply:
Added MultiIndex and non-unique tests